src/overlays/actors/ovl_Demo_Im/z_demo_im.c

```diff
@@ -21,229 +94,1125 @@ const ActorInit Demo_Im_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/func_809871B4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/func_809871E8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/func_80987288.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/func_809872A8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/func_809872F0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/func_80987330.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/DemoIm_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/DemoIm_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/DemoIm_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/func_80987514.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/func_809875C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/func_8098764C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/func_80987658.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Im/DemoIm_Draw.s")
+extern UNK_PTR D_02002524;
+extern AnimationHeader D_06000710;
+extern AnimationHeader D_06000AFC;
+extern Gfx D_0600EDE8[];
+extern AnimationHeader D_060014E4;
+extern AnimationHeader D_06001868;
+extern SkeletonHeader D_0600F788;
+extern AnimationHeader D_0600FB10;
+extern AnimationHeader D_060101C8;
+extern AnimationHeader D_06010EE0;
+extern AnimationHeader D_0601182C;
+extern AnimationHeader D_06011C08;
+extern AnimationHeader D_06012218;
+
+void func_80984BE0(DemoIm* this) {
+    s32 pad[3];
+    s16* unk_25E = &this->unk_25E;
+    s16* unk_25C = &this->unk_25C;
+
+    if (DECR(*unk_25E) == 0) {
+        *unk_25E = Math_Rand_S16Offset(0x3C, 0x3C);
+    }
+
+    *unk_25C = *unk_25E;
+    if (*unk_25C >= 3) {
+        *unk_25C = 0;
+    }
+}
+
+void func_80984C68(DemoIm* this) {
+    this->action = 7;
+    this->drawConfig = 0;
+    this->unk_26C = 0;
+    this->unk_270 = 0;
+    this->actor.shape.unk_14 = 0;
+    this->unk_268 = 0.0f;
+}
+
+void func_80984C8C(DemoIm* this, GlobalContext* globalCtx) {
+    u32* something = &D_8098783C;
+
+    if (globalCtx->csCtx.state == 0) {
+        if (*something != 0) {
+            if (this->actor.params == 2) {
+                func_80984C68(this);
+            }
+            *something = 0;
+        }
+    } else {
+        if (*something == 0) {
+            *something = 1;
+        }
+    }
+}
+
+void func_80984D00(DemoIm* this, GlobalContext* globalCtx) {
+    ColliderCylinder* collider = &this->collider;
+    if (this) {} // needed to match
+
+    Collider_InitCylinder(globalCtx, collider);
+    Collider_SetCylinder_Set3(globalCtx, collider, &this->actor, &sCylinderInit);
+}
+
+void func_80984D4C(DemoIm* this, GlobalContext* globalCtx) {
+    ColliderCylinder* collider = &this->collider;
```

Comment:
Similar thing here, the collider temp is very unlikely to be a real thing, but this function could very well take `Actor* thisx`